Cumulus test service cleanup

skunert opened the pull request:
closes #2567 

Followup for https://github.com/paritytech/polkadot-sdk/pull/2331

This PR contains multiple internal cleanups:

1. This gets rid of the functionality in `generate_genesis_block` which was only used in one benchmark
2. Fixed `transaction_pool` and `transaction_throughput` benchmarks failing since they require a tokio runtime now.
3. Removed `parachain_id` CLI option from the test parachain
4. Removed `expect` call from `RuntimeResolver`

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4874545

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4876828

michalkucharczyk commented on cumulus/client/cli/src/lib.rs:
```diff
@@ -127,6 +127,27 @@ impl sc_cli::CliConfiguration for PurgeChainCmd {
+			.hash(Zero::zero())?
+			.ok_or(sc_cli::Error::Client(sp_blockchain::Error::Backend(
+				"Failed to lookup genesis block hash when exporting genesis head data.".into(),
+			)))?;
```
maybe `client.chain_info().genesis_hash` is available here?

skunert commented:
@pepoviola In this PR I removed the `parachain-id` CLI flag from the test collator. In addition, I removed the option to do let out `--chain`, because the test-parachain by default uses the test runtime which is not used by zombienet.

However tests are not failing while trying to start `temp-3`. To me its unclear what this is supposed to do, looks like a leftover somehow. Command for temp-3
```
bash -c test-parachain build-spec  --disable-default-bootnode > /cfg/undefined-plain.json && echo done > /tmp/zombie-tmp-done && until [ -f /tmp/finished.txt ]; do echo waiting for copy files to finish; sleep 1; done; echo copy files has finished 
```

It is writing to `undefined-plain.json` and now fails since not `--chain` is provided. WDYT?

pepoviola commented:
Hi @skunert, thanks for the head up. I will take a look asap, and this may require a new version to support this changes.
Thx!
